components/brave_vpn/resources/panel/components/toggle/index.tsx

```diff
@@ -1,22 +1,56 @@
 import * as React from 'react'
+import { LoaderIcon } from 'brave-ui/components/icons'
+
 import * as S from './style'
+import locale from '../../constants/locale'
+import { ConnectionState } from '../../api/panel_browser_api'
+import { useSelector, useDispatch } from '../../state/hooks'
+import * as Actions from '../../state/actions'
 
-interface Props {
-  isOn: boolean
-  onClick: () => void
+function useIsOnSelector () {
+  // We derive isOn from connectionStatus to be synchronized
+  const connectionStatus = useSelector(state => state.connectionStatus)
+  const isConnectionActive = [ConnectionState.CONNECTED, ConnectionState.CONNECTING].includes(connectionStatus)
+  return isConnectionActive
 }
```

Comment:
follow-up: put this in `state/selectors` file or directory if this is going to be needed to be read elsewhere